synapse/lib/stormhttp.py

```diff
@@ -464,12 +464,27 @@ async def _httpRequest(self, meth, url, headers=None, json=None, body=None,
                     kwargs['json'] = json
 
                 async with sess.request(meth, url, headers=headers, **kwargs) as resp:
+                    history = []
+                    for hist in resp.history:
+                        hnfo = {
+                            'code': hist.status,
+                            'reason': await self.codereason(hist.status),
+                            'headers': dict(hist.headers),
+                            'url': str(hist.url),
+                            # aiohttp has already closed the connection by this point
+                            # so there is no connection to read a body from.
+                            'body': b'',
+                            'history': [],
+                            'request_headers': dict(hist.request_info.headers)
+                        }
+                        history.append(hnfo)
                     info = {
                         'code': resp.status,
                         'reason': await self.codereason(resp.status),
                         'headers': dict(resp.headers),
                         'url': str(resp.url),
                         'body': await resp.read(),
+                        'history': history,
```

Comment:
I did debate treating this like a linked list of responses vs a flat list. The main difference is really how you would access and process them via Storm. It felt more friendly for a `gtor` to return a [ potentially empty ] list than have a `gtor` that returned either `None` or `inet:http:resp`. With a flag list you can know the # of redirects right away; but with linked list you have to walk them until you have no more history and that felt odd.

Reply:
Yeah a linked list feels like it would be clunky to work with in storm